src/components/MovedStopHandler.tsx

```diff
@@ -0,0 +1,101 @@
+import React, {FC, useContext} from 'react';
+import Button from '@mui/material/Button';
+
+import {MapContext} from './contexts/MapContextProvider';
+
+const MovedStopHandler: FC = () => {
+  const {activeStop, movedStopsState, resetPositionFunction} = useContext(MapContext);
+
+  const saveStopToLocalStorage = () => {
+    if (!activeStop) {
+      return;
+    }
+    const newItem = movedStopsState.find(({id}) => id === activeStop.id);
+    if (newItem) {
+      const storage = localStorage.getItem('moveStopsArray');
+      if (storage) {
+        try {
+          const storageObject = JSON.parse(storage);
+          const index = storageObject.findIndex(({id}: {id: string}) => id === activeStop.id);
+          console.log({index});
+          if (index >= 0) {
+            storageObject[index] = {
+              id: storageObject[index].id,
+              externalId: storageObject[index].externalId,
+              position: newItem.position,
+            };
+            console.log({storageObject});
+
+            localStorage.setItem('moveStopsArray', JSON.stringify(storageObject));
+          } else {
+            const newArray = storageObject.concat([newItem]);
+            localStorage.setItem('moveStopsArray', JSON.stringify(newArray));
+          }
+        } catch {
+          console.log('Wystąpił błąd przy pobieraniu danych');
```

Comment:
Maybe it would be a good idea to use NotificationActions, so user is notified about this error and can give us a hint while reporting a bug?